godef.go

```diff
@@ -225,7 +227,24 @@ func (o orderedObjects) Swap(i, j int)      { o[i], o[j] = o[j], o[i] }
 func done(obj *ast.Object, typ types.Type) {
 	defer os.Exit(0)
 	pos := types.FileSet.Position(types.DeclPos(obj))
-	fmt.Printf("%v\n", pos)
+	if *jsonFlag {
+		p := struct {
+			Filename string `json:"filename,omitempty"`
+			Line     int    `json:"line,omitempty"`
+			Column   int    `json:"column,omitempty"`
+		}{
+			Filename: pos.Filename,
+			Line:     pos.Line,
+			Column:   pos.Column,
+		}
+		jsonStr, err := json.Marshal(p)
+		if err != nil {
+			fail("JSON marshal error: %v", err)
+		}
+		fmt.Printf("%s\n", jsonStr)
```

Comment:
Ah, I've just realised that this isn't sufficient, as there are other
places where it prints output below if the -t flag is specified.
So if you specify -t, you won't get valid JSON output.
Perhaps for simplicity we should just return here and change
the flag description to:

     "output location in JSON format (-t flag is ignored)"